src/celestia/celx_phase.cpp

```diff
@@ -98,7 +73,7 @@ static int phase_timespan(lua_State* l)
 
     celx.checkArgs(1, 1, "No arguments allowed for to phase:timespan");
 
-    const TimelinePhase* phase = this_phase(l);
+    auto& phase = *this_phase(l);
```

Comment:
if it possible to put copy of shared_ptr to luauserdata? if not then it's better to put a raw pointer to data instead of the raw pointer to a smart pointer

Reply:
That's how it's implemented already. Lua userdata is always presented to user as a raw pointer by default. So, by default we always have raw pointer to smart pointer. :-)

Reply:
okay